### PR TITLE
(GH-137) Add TAR and GZIP utility functions

### PR DESCRIPTION
**Pull request by sanfrancrisko**
This commit adds utility functions to allow directories to be TAR
and GZIP'd. There are also functions to extract the contents of a
TAR and GZIP archive too.

This is in preparation for the upcoming template packaging work.

Closes: #137

### CODE REVIEW

File: internal/pkg/pkg_template/package_template.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+var gzipUtil gzipHelpers
+var ioUtil ioUtils
+
+func init() {
+	tarUtil = tarHelpersImpl{}
+	ioUtil = ioUtilHelpersImpl{}
+	gzipUtil = gzipHelpersImpl{}
+}
+
+func PackageTemplate(templatePath string) (archiveFilePath string, err error) {
```
Comment (jpogran) [inline internal/pkg/pkg_template/package_template.go:20]: I don't see the need for this to be a separate package. This is directly relevant to the `pct` package in that it builds and packages templates, so it should be over there and not in a new package. As a bonus you won't have to duplicate test files and other things. 

The gzip and tar can be their own packages, as they are outside concerns to the main goal.

In summary, create a new file called `pct/build.go` and put `package_template.go` content there.

File: internal/pkg/gzip/gunzip.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import (
+	"compress/gzip"
+	"os"
+	"path/filepath"
+
+	"github.com/puppetlabs/pdkgo/internal/pkg/utils"
+)
+
+func Gunzip(source, target string) error {
```
Comment (jpogran) [inline internal/pkg/gzip/gunzip.go:11]: Do we need gunzip and gzip?

**Review by jpogran — CHANGES_REQUESTED** [review]
Great work so far!

File: internal/pkg/gzip/gunzip.go (hunk 2 above)
Comment (sanfrancrisko) [reply internal/pkg/gzip/gunzip.go:12]: Just for the benefit of those viewing, the `Gunzip` is the method that'll extract a `.tar.gz` packaged template

File: internal/pkg/gzip/gzip_test.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var gzipOutputPath string
+
+func getGzipOutputPath() string {
+	gzipOutputPath, _ = ioutil.TempDir("", "")
```
Comment (jpogran) [inline internal/pkg/gzip/gzip_test.go:13]: Same comment as gunzip test method

File: internal/pkg/tar/tar_test.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+)
+
+func TestTar(t *testing.T) {
+
+	tempDir, _ := ioutil.TempDir("", "")
```
Comment (jpogran) [inline internal/pkg/tar/tar_test.go:11]: Another for t.TempDir

File: internal/pkg/tar/untar_test.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestUntar(t *testing.T) {
+
+	tempDir, _ := ioutil.TempDir("", "")
```
Comment (jpogran) [inline internal/pkg/tar/untar_test.go:12]: Another for t.TemptDir

**Review by jpogran — APPROVED** [review]

**Pull request merged**